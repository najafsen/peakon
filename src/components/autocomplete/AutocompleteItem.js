import React from 'react';
import cx from 'classnames';
import { Avatar } from '../avatar/Avatar';

export const AutocompleteItem = ({ item, isActive, onSelect, onMouseOver }) => {
    return (
        <li
            data-testid="li"
            className={cx('autocomplete-item', { 'active': isActive })}
            onClick={() => onSelect(item)}
            onMouseOver={() => onMouseOver(item)}
        >
            <div className="item-content">
                <div className="avatar">
                    <Avatar src={item.avatar} name={item.name} />
                </div>
                <div className="info">
                    <div className="name">{item.name}</div>
                    <div className="email">{item.email}</div>
                </div>
            </div>
        </li>
    );
}