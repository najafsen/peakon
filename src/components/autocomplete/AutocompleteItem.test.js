import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { AutocompleteItem } from './AutocompleteItem';

describe('AutocompleteItem', () => {
    let item, isActive, onSelect, onMouseOver;

    beforeEach(() => {
        item = {
            name: 'The Name',
            email: 'the@email',
            avatar: 'the/src',
        };
        onSelect = jest.fn();
        onMouseOver = jest.fn();
        isActive = false;
    });

    test('default matches snapshot', () => {
        const { asFragment } = render(<AutocompleteItem item={item} isActive={isActive} onSelect={onSelect} onMouseOver={onMouseOver} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('active matches snapshot', () => {
        const { asFragment } = render(<AutocompleteItem item={item} isActive={true} onSelect={onSelect} onMouseOver={onMouseOver} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('default does not have active class', () => {
        const { getByTestId } = render(<AutocompleteItem item={item} isActive={isActive} onSelect={onSelect} onMouseOver={onMouseOver} />);
        expect(getByTestId('li')).not.toHaveClass('active');
    });

    test('active has active class', () => {
        const { getByTestId } = render(<AutocompleteItem item={item} isActive={true} onSelect={onSelect} onMouseOver={onMouseOver} />);
        expect(getByTestId('li')).toHaveClass('active');
    });

    test('onSelect event', () => {
        const { getByTestId } = render(<AutocompleteItem item={item} isActive={true} onSelect={onSelect} onMouseOver={onMouseOver} />);
        fireEvent.click(getByTestId('li'));
        expect(onSelect).toHaveBeenCalledWith(item);
    });

    test('onMouseOver event', () => {
        const { getByTestId } = render(<AutocompleteItem item={item} isActive={true} onSelect={onSelect} onMouseOver={onMouseOver} />);
        fireEvent.mouseOver(getByTestId('li'));
        expect(onMouseOver).toHaveBeenCalledWith(item);
    });

    // More tests should be added, because of lack of time I wrote only these sample tests
});