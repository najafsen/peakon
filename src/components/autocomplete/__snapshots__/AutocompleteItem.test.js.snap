// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteItem active matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="autocomplete-item active"
    data-testid="li"
  >
    <div
      class="item-content"
    >
      <div
        class="avatar"
      >
        <div
          class="avatar-component"
        >
          <div
            class="avatar-alt"
            data-testid="alt"
          >
            TN
          </div>
          <img
            alt="The Name"
            class="avatar-image hidden"
            data-testid="image"
            src="the/src"
          />
        </div>
      </div>
      <div
        class="info"
      >
        <div
          class="name"
        >
          The Name
        </div>
        <div
          class="email"
        >
          the@email
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;

exports[`AutocompleteItem default matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="autocomplete-item"
    data-testid="li"
  >
    <div
      class="item-content"
    >
      <div
        class="avatar"
      >
        <div
          class="avatar-component"
        >
          <div
            class="avatar-alt"
            data-testid="alt"
          >
            TN
          </div>
          <img
            alt="The Name"
            class="avatar-image hidden"
            data-testid="image"
            src="the/src"
          />
        </div>
      </div>
      <div
        class="info"
      >
        <div
          class="name"
        >
          The Name
        </div>
        <div
          class="email"
        >
          the@email
        </div>
      </div>
    </div>
  </li>
</DocumentFragment>
`;
